### PR TITLE
[WebGPU] Update to latest native headers

### LINKED ISSUE #7581: The process for generating `mini_webgpu.h` need documenting
These structs (etc) are generated from a json file in the Dawn source code; apparently the technique for extracting what we need isn't entirely trivial, so we should document it somewhere in source.

attn @jrprice

### PR DESCRIPTION
**Pull request by jrprice**
* Update mini_webgpu.h with latest version from Dawn
  * Document this process
* Remove an argument from wgpuQueueOnSubmittedWorkDone

Fixes #7581

### CODE REVIEW

File: README_webgpu.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ JIT compilation is not supported when using WebGPU with WASM.
 
 ## Running natively: `HL_TARGET=host-webgpu`
 
-> _Tested with top-of-tree Dawn as of 2023-03-14._
+> _Tested with top-of-tree Dawn as of 2023-11-03._
```
Comment (shoaibkamil) [inline README_webgpu.md:50]: Can you add a short line in this section (perhaps before line 55?) that is essentially:
```
Note that as of 2023-10-31,
wgpu is not supported due to [lacking `override` support for wgsl](https://github.com/gfx-rs/wgpu/issues/1762)
which we require in order to set GPU block sizes.```

File: README_webgpu.md [hunk 2]
```diff
@@ -108,3 +108,16 @@ This will produce the following artifacts:
...
+The recommended method for updating `mini_webgpu.h` is to copy the
+`gen/include/dawn/webgpu.h` file from the Dawn build directory, then:
+- Restore the `// clang-format {off,on}` lines.
+- Comment out the `#include <std*>` lines.
+- Remove the `void` parameter from the `WGPUProc` declaration.
+
+This guarantees a version of the WebGPU header that is compatible with Dawn.
+When the native API eventually stabilizes, it should be possible to obtain a
+header from the `webgpu-native` GitHub organization that will be compatible
+with Dawn, wgpu, and Emscripten.
```
Comment (shoaibkamil) [inline README_webgpu.md:123]: Drive-by request for a change in this file: can we make all the links have `https://` in front of them?  Otherwise they get parsed as local links.

**Review by shoaibkamil — APPROVED** [review]
Beat me to it :) We need to update the dawn implementations on the buildbots, but I'll coordinate that with @abadams.  Thanks for doing this!

File: README_webgpu.md (hunk 2 above)
Comment (jrprice) [reply README_webgpu.md:128]: Done.

File: README_webgpu.md (hunk 1 above)
Comment (jrprice) [reply README_webgpu.md:50]: Done.

**Comment by abadams** [comment]
I've updated dawn on both of the x86 mac buildbots we use for testing it. Doing a merge with main should clear up the buildbot failures.

**Comment by jrprice** [comment]
> Doing a merge with main should clear up the buildbot failures.

Done.

**Comment by abadams** [comment]
Looks like something's up with the async_device_copy correctness test with target host-webgpu

**Comment by steven-johnson** [comment]
I'm updating Dawn (again) to the latest headers as the failing armbot had a stale version.
